### PR TITLE
Add a PR template for refactors

### PR DESCRIPTION
**Pull request by tibetiroka**
Adds a new PR template for PRs that refactor something. The style is in line with the rest of the PR templates.

### CODE REVIEW

File: .github/pull_request_template.md [hunk 1]
```diff
@@ -65,3 +65,20 @@ This save file can be used to test the balance changes.
 
 ## Performance Impact
 {{describe any performance impact (positive or negative). "N/A" if no performance-critical code is changed. }}
+
+
+-----------------------
+**Refactor:** This PR refactors {{insert file name}}.
+
+## Reasoning
+{{explain why you thing the refactor was necessary, and how it is an improvement}}
```
Comment (Quantumshark) [inline .github/pull_request_template.md:74]:
```suggestion
{{explain why you think the refactor was necessary, and how it is an improvement}}
```

File: .github/pull_request_template.md [hunk 2]
```diff
@@ -65,3 +65,20 @@ This save file can be used to test the balance changes.
 
 ## Performance Impact
 {{describe any performance impact (positive or negative). "N/A" if no performance-critical code is changed. }}
+
+
+-----------------------
+**Refactor:** This PR refactors {{insert file name}}.
```
Comment (Saugia) [inline .github/pull_request_template.md:71]:
```suggestion
**Refactor:** This PR refactors {{insert file name(s)}}.
```
Very minor, but just in case.

**Review by Quantumshark — APPROVED** [review]

**Comment by quyykk** [comment]
I'm not entirely convinced we need a template for this. The primary issue for me is that GitHub doesn't provide proper UX for selecting templates, so every template we add just increases the amount of text to delete and might make it harder for people to find templates? idk

**Comment by tibetiroka** [comment]
Yeah, GitHub really doesn't support PR templates well.

**Comment by tibetiroka** [comment]
I don't know if we _need_ this. It's just annoying that there isn't a template I can fill out for refactors ~~and then I get yelled at for not using the templates~~.

**Comment by quyykk** [comment]
Just use the feature one I guess, deleting the other sections that don't apply

**Pull request closed without merging**